libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+                    break;
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static buildMessageActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Message };
+        for (const key of Object.keys(lgJObj)) {
+            const property: string = key.trim();
+            const value: any = lgJObj[key];
+
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'text':
+                    activity.text = value.toString();
+                    break;
+                case 'speak':
+                    activity.speak = value.toString();
+                    break;
+                case 'inputhint':
+                    activity.inputHint = value.toString();
+                    break;
+                case 'attachments':
+                    activity.attachments = this.getAttachments(value);
+                    break;
+                case 'suggestedactions':
+                    activity.suggestedActions = this.getSuggestions(value);
+                    break;
+                case 'attachmentlayout':
+                    activity.AttachmentLayout = value.toString();
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static getSuggestions(value: any): SuggestedActions {
+        let suggestedActions: SuggestedActions = {
+            actions : [],
+            to: []
+        };
+
+        let actions: any[] = this.normalizedToList(value);
+
+        for (const action of actions) {
+            if (typeof action === 'string') {
+                const actionStr: string = action.trim();
+                suggestedActions.actions.push({
+                    type: ActionTypes.MessageBack,
+                    title: actionStr,
+                    displayText: actionStr,
+                    text: actionStr,
+                    value: actionStr
+                });
+            } else {
+                const cardAction: CardAction = this.getCardAction('', action);
+                if (cardAction !== undefined) {
+                    suggestedActions.actions.push(cardAction);
+                }
+            }
+        }
+
+        return suggestedActions;
+    }
+
+    private static getButtons(cardType: string, value: any): CardAction[] {
+        let buttons: any[] = [];
+        let actions: any[] = this.normalizedToList(value);
+
+        for (const action of actions) {
+            if (typeof action === 'string') {
+                const actionStr = action.trim();
+                if (cardType === CardFactory.contentTypes.signinCard ||cardType === CardFactory.contentTypes.oauthCard ) {
```

Comment:
>cardType === CardFactory.contentTypes.signinCard ||cardType === CardFactory.contentTypes.oauthCard  [](start = 20, length = 99)

This logic would make every button for a sign-in card a sign-in button. That doesn't seem desirable.